cpp/src/text/subword/load_hash_file.cu

```diff
@@ -293,10 +262,17 @@ std::unique_ptr<hashed_vocabulary> load_vocabulary_file(
                            cudaMemcpyHostToDevice,
                            stream.value()));
 
-  // this just initializes some constant tables into device memory
-  // to help speed up the runtime
-  detail::get_codepoint_metadata(stream);
-  detail::get_aux_codepoint_data(stream);
+  auto cp_metadata = detail::get_codepoint_metadata(stream);
+  result.cp_metadata =
```

Comment:
We have a latent undefined behavior in the function call here. We are presuming a Left to Right evaluation of the arguments to the function which by https://en.cppreference.com/w/cpp/language/eval_order is undefined.

So if the evaluation order becomes Right to Left, `release()` will occur before `size()` and we will have the wrong size

Reply:
Oh right, the order of argument evaluation is not defined. Sorry I missed this. So the `size` need to be recorded before release.